cmd/flakechecker/bq_client.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"cloud.google.com/go/bigquery"
+	"context"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+	"time"
+)
+
+const projectID = "acs-san-stackroxci"
+const queryTimeout = 1 * time.Minute
+const queryStrGetFailureRatio = `
+SELECT
+    JobName,
+    FilteredName,
+    Classname,
+    TotalAll,
+    FailRatio
+FROM
+` + "`acs-san-stackroxci.ci_metrics.stackrox_tests_recent_flaky_tests`" + `
+WHERE
+    JobName = @jobName
+    AND FilteredName = @filteredName
+    AND Classname = @classname
+`
+
+type biqQueryClient interface {
+	GetRatioForTest(flakeTestConfig *flakeCheckerRecord, testName string) (int, int, error)
+}
+
+type biqQueryClientImpl struct {
+	client *bigquery.Client
+}
+
+func getNewBigQueryClient() (biqQueryClient, error) {
+	ctx := context.Background()
```

Comment:
Timeout for getting a client too?

Reply:
I have checked the documentation a bit: https://pkg.go.dev/cloud.google.com/go#hdr-Timeouts_and_Cancellation
They are suggesting not to set timeouts for `NewClient`.
> Do not attempt to control the initial connection (dialing) of a service by setting a timeout on the context passed to NewClient. Dialing is non-blocking, so timeouts would be ineffective and would only interfere with credential refreshing, which uses the same context.